integration_tests/src/main/python/array_test.py

```diff
@@ -394,3 +394,80 @@ def test_array_max_q1():
     def q1(spark):
         return spark.sql('SELECT ARRAY_MAX(TRANSFORM(ARRAY_REPEAT(STRUCT(1, 2), 0), s -> s.col2))')
     assert_gpu_and_cpu_are_equal_collect(q1)
+
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen,
+    FloatGen(special_cases=[]), DoubleGen(special_cases=[]), string_gen, boolean_gen, date_gen, timestamp_gen], ids=idfn)
+def test_array_intersect(data_gen):
+    gen = StructGen(
+        [('a', ArrayGen(data_gen, nullable=False)),
+        ('b', ArrayGen(data_gen, nullable=False))],
```

Comment:
I see that all the new tests don't have nullable input?

Reply:
Spark doesn't support either parameter being null from what I can tell

Reply:
The type check is performed by Spark itself (even the CPU doesn't even take null inputs to these functions). 

Here is spark shell from base Spark (no plugin loaded).

```
scala> spark.sql("select array_union(NULL,array(1))")
org.apache.spark.sql.AnalysisException: cannot resolve 'array_union(NULL, array(1))' due to data type mismatch: input to function array_union should have been two arrays with same element type, but it's [void, array<int>]; line 1 pos 7;
```

Reply:
Here is my test that works:
```
scala> val df1 = Seq((Seq(1, 2), Seq(3, 2)), (null, Seq(4,5))).toDF("a", "b")
df1: org.apache.spark.sql.DataFrame = [a: array<int>, b: array<int>]

scala> df1.show
+------+------+
|     a|     b|
+------+------+
|[1, 2]|[3, 2]|
|  null|[4, 5]|
+------+------+


scala> df1.selectExpr("arrays_overlap(a, b)").show
+--------------------+
|arrays_overlap(a, b)|
+--------------------+
|                true|
|                null|
+--------------------+
```
So your example above may need some explicit way to convert `NULL` into an array type.